src/app/unlocked-stage/unlocked-stage.component.ts

```diff
@@ -1,156 +1,331 @@
-    public language: any;
-
-    constructor(private _langService: LanguageService, private _renderer: Renderer, private _dashboardService: DashboardService, private _sharedData: SharedDataService) { }
-
-    ngOnInit() {
-        this._langService.loadLanguage().subscribe(response => {
-            this.language = response.pcprepkit.stages.lockedStage.game;
-        });
-        this._humanProgBar = document.getElementById('human-health-bar');
-        this._mosquitoProgBar = document.getElementById('mosquito-health-bar');
-        this.mosquito = document.getElementById('moz');
-        this._human = document.getElementById('human');
-        this._sprayItem = document.getElementById('spray-item');
-        this._mousedownListener = this._renderer.listen(this._sprayItem, 'click', (event) => this.changeCursor(event));
-        this.animateDiv();
-    }
+  public gameOver = false;
+  public success = false;
+  public language: any;
+  public time = 0;
+  public spray: boolean;
+  public net: boolean;
+  public sprayValue = 100;
+  public starCount = 0;
 
-    changeCursor(e) {
-        this._spray = document.getElementById('spray');
-        this._canvas =  document.getElementById('scene');
-        this._mousemoveListener = this._renderer.listen(this._canvas, 'mousemove', (event) => this.moveOverCanvas(event));
-        this._mouseleaveListener = this._renderer.listen(this._canvas, 'mouseleave', (event) => this.leaveCanvas());
-        this._mousedownListener = this._renderer.listen(this._canvas, 'mousedown', (event) => this.attack());
-        this._mouseupListener = this._renderer.listen(this._canvas, 'mouseup', (event) => this.stopAttack());
-    }
+  private _canvas;
+  private _spray;
+  private $human;
+  private _humanProgBar;
+  private _collisionWithManStatus = false;
+  private _collisionWithNetStatus = false;
+  private _humanBarWidth = 100;
+  private _mousemoveListener;
+  private _mouseleaveListener;
+  private _mousedownListener;
+  private _mouseupListener;
+  private _mousedown = false;
+  private _status: object = {stage: 5, activity: 3};
+  private _timeout;
+  private $mosquito;
+  private $spray;
+  private $net;
+  private mosquitoes = [];
+  private maxMosquitoes = 10;
+  private mosquitoesStrength = Array(this.maxMosquitoes).fill(100);
+  private mosCountInitial = 2;
+  private mosCountCurrent = this.mosCountInitial;
+  private keydown = false;
+  private levelCounter = 1;
+  private maxLevel = 4;
+
+  constructor(private _langService: LanguageService, private _renderer: Renderer2, private _dashboardService: DashboardService, private _sharedData: SharedDataService,
+              public snackBar: MatSnackBar) {
+  }
 
-    attack() {
-        this._spray.src = '../assets/img/unlocked-stage/spray.gif';
-        this._mousedown = true;
+  // If we need keyboard movement for the net
+  @HostListener('window:keyup', ['$event'])
+  keyEvent(event: KeyboardEvent) {
+
+    if (event.keyCode === 38) {
```

Comment:
The keycodes can be made constants for better readability

Reply:
Done.